### PR TITLE
Use new publishing plugin, some refactoring.

### PR DESCRIPTION
**Pull request by mattiahj**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -311,15 +298,15 @@ task mavenCentralSubprojects {
...
-    dependsOn ':secrets:publish'
-    finalizedBy ':closeAndReleaseRepository'
+    dependsOn ':apigateway:publishToSonatype'
+    dependsOn ':core:publishToSonatype'
+    dependsOn ':doi:publishToSonatype'
+    dependsOn ':eventhandlers:publishToSonatype'
+    dependsOn ':identifiers:publishToSonatype'
+    dependsOn ':lambdaauthorizer:publishToSonatype'
+    dependsOn ':logutils:publishToSonatype'
+    dependsOn ':secrets:publishToSonatype'
```
Comment (axthosarouris) [inline build.gradle:308]: dependsOn 's3:publishToSonatype'

File: build.gradle [hunk 2]
```diff
@@ -294,34 +281,6 @@ task jacocoRootReport(type: JacocoReport, dependsOn: tasks.jacocoMerge) {
...
-    def proc = "git rev-parse --abbrev-ref HEAD".execute()
-    proc.in.eachLine { line -> branch = line }
-    proc.err.eachLine { line -> println line }
-    proc.waitFor()
-    branch
-}
-
-task mavenCentralSubprojects {
-    def branch = gitBranch()
-    if (!(branch == "master" || branch == "main")) {
```
Comment (mattiahj) [inline build.gradle:308]: The github actions already only runs on master and main

File: build.gradle [hunk 3]
```diff
@@ -294,34 +281,6 @@ task jacocoRootReport(type: JacocoReport, dependsOn: tasks.jacocoMerge) {
...
-        logger.warn("Warning: Can only publish to Maven Central from master or main branch")
-        return
-    }
-
-    // define which subprojects to publish to Maven Central
-    dependsOn ':apigateway:publish'
```
Comment (mattiahj) [inline build.gradle:314]: Publish all maven publications

File: build.gradle (hunk 2 above)
Comment (axthosarouris) [reply build.gradle:308]: I think this is a security check in case someone misconfigures github actions.

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**